org/mozilla/jss/ssl/javax/JSSEngine.java

```diff
@@ -249,6 +259,20 @@ protected static String errorText(int error) {
         }
     }
 
+    /**
+     * Safely initializes the session cache if not already initialized.
+     */
+    public static void initializeSessionCache(
+        int maxCacheEntries, long timeout, String directory)
+    {
+        if (sessionCacheInitialized.compareAndSet(false, true)) {
+            if (SSL.ConfigServerSessionIDCache(maxCacheEntries, timeout, timeout, directory) == SSL.SECFailure) {
+                String msg = "Unable to configure server session cache: ";
+                msg += errorText(PR.GetError());
+                throw new RuntimeException(msg);
```

Comment:
Could we throw an `SSLException` instead and declare it in this method? The `JSSEngineReferenceImpl.beginHandshake()` should have declared an `SSLException` so we can just let it bubble up.

Generally I'd rather not create a `RuntimeException` as the initial exception unless we can't declare the exception in the method.

Reply:
Sure, I think that's possible. I think originally `beginHandshake()` was called from `updateHandshakeState()` which in turn is always called from `getHandshakeStatus()` -- the latter declares no exceptions.

Reply:
Yeah we just need to catch one location of `SSLException` and re-throw as `RuntimeException`, but otherwise we can handle that better now.